src/main/scala/com/cognite/spark/datasource/DataPointsRelationV1.scala

```diff
@@ -0,0 +1,299 @@
+      case GreaterThan("timestamp", value) => Seq(timeStampStringToMin(value))
+      case GreaterThanOrEqual("timestamp", value) => Seq(timeStampStringToMin(value))
+      case And(f1, f2) => getTimestampLimit(f1) ++ getTimestampLimit(f2)
+      // case Or(f1, f2) => we might possibly want to do something clever with joining an "or" clause
+      //                    with timestamp limits on each side (including replacing "max of Min's" with the less strict
+      //                    "min of Min's" when aggregating filters on the same side); just ignore them for now
+      case _ => Seq.empty
+    }
+
+  def timeStampStringToMin(value: Any): Min =
+    Min(java.sql.Timestamp.valueOf(value.toString).toInstant.plusMillis(1).toEpochMilli)
+
+  def timeStampStringToMax(value: Any): Max =
+    Max(java.sql.Timestamp.valueOf(value.toString).toInstant.minusMillis(1).toEpochMilli)
+
+  // scalastyle:off cyclomatic.complexity
+  def getNameFilters(filter: Filter): Seq[NameFilter] =
+    filter match {
+      case IsNotNull("name") => Seq()
+      case EqualTo("name", value) => Seq(NameFilter(value.toString))
+      case EqualNullSafe("name", value) => Seq(NameFilter(value.toString))
+      case In("name", values) => values.map(v => NameFilter(v.toString))
+      case And(f1, f2) => getNameFilters(f1) ++ getNameFilters(f2)
+      case Or(f1, f2) => getNameFilters(f1) ++ getNameFilters(f2)
+      case StringStartsWith("name", value) =>
+        // TODO: add support for this using the "q" parameter when listing time series
+        sys.error(
+          s"Filtering using 'string starts with' not allowed for data points, attempted for ${value.toString}")
+      case StringEndsWith("name", value) =>
+        // TODO: add support for this using the time series search endpoint
+        sys.error(
+          s"Filtering using 'string ends with' not allowed for data points, attempted for ${value.toString}")
+      case StringContains("name", value) =>
+        // TODO: add support using the time series search endpoint
+        sys.error(
+          s"Filtering using 'string contains' not allowed for data points, attempted for ${value.toString}")
+      case _ =>
+        Seq()
+    }
+  // scalastyle:on cyclomatic.complexity
+
+  private def getAggregationSettings(
+      filters: Array[Filter]): (Array[AggregationFilter], Seq[String]) = {
+    val aggregations = filters.flatMap(getAggregation).distinct
+    val granularities = filters.flatMap(getGranularity).distinct
+
+    if (aggregations.nonEmpty && granularities.isEmpty) {
+      sys.error(s"Aggregations requested but granularity is not specified")
+    }
+
+    if (aggregations.isEmpty && granularities.nonEmpty) {
+      sys.error(s"Granularity specified but no aggregation requested")
+    }
+
+    (aggregations, granularities)
+  }
+
+  // scalastyle:off cyclomatic.complexity
+  def toAggregationFilter(aggregation: String): AggregationFilter =
```

Comment:
Abbreviations like `avg`, `cv` etc. are no longer supported in V1, so this can (and must be) be simplified.